contracts/token/Token.sol

```diff
@@ -19,25 +20,75 @@ contract Token is IToken, TransferManager {
         string memory _version,
         address _onchainID)
     public TransferManager(_identityRegistry, _compliance) {
-        name = _name;
-        symbol = _symbol;
-        decimals = _decimals;
-        version = _version;
-        onchainID = _onchainID;
+        tokenName = _name;
+        tokenSymbol = _symbol;
+        tokenDecimals = _decimals;
+        tokenVersion = _version;
+        tokenOnchainID = _onchainID;
+
+        emit UpdatedTokenInformation(tokenName, tokenSymbol, tokenDecimals, tokenVersion, tokenOnchainID);
+    }
+
+    /**
+     * @dev Returns the number of decimals used to get its user representation.
+     * For example, if `decimals` equals `2`, a balance of `505` tokens should
+     * be displayed to a user as `5,05` (`505 / 10 ** 2`).
+     *
+     * Tokens usually opt for a value of 18, imitating the relationship between
+     * Ether and Wei.
+     *
+     * NOTE: This information is only used for _display_ purposes: it in
+     * no way affects any of the arithmetic of the contract, including
+     * balanceOf() and transfer().
+     */
+    function decimals() public override view returns (uint8){
+        return tokenDecimals;
+    }
+
+    /**
+     * @dev Returns the name of the token.
+     */
+    function name() public override view returns (string memory){
+        return tokenName;
+    }
 
-        emit UpdatedTokenInformation(name, symbol, decimals, version, onchainID);
+    /**
+     * @dev Returns the address of the onchainID of the token.
+     * the onchainID of the token gives all the information available
+     * about the token and is managed by the token issuer or his agent.
+     */
+    function onchainID() public override view returns (address){
+        return tokenOnchainID;
     }
 
     /**
-    * Owner can update token information here
-    */
-    function setTokenInformation(string calldata _name, string calldata _symbol, uint8 _decimals, string calldata _version, address _onchainID) external onlyOwner {
-        name = _name;
-        symbol = _symbol;
-        decimals = _decimals;
-        version = _version;
-        onchainID = _onchainID;
-
-        emit UpdatedTokenInformation(name, symbol, decimals, version, onchainID);
+     * @dev Returns the symbol of the token, usually a shorter version of the
+     * name.
+     */
+    function symbol() public override view returns (string memory){
+        return tokenSymbol;
+    }
+
+    /**
+     * @dev Returns the TREX version of the token.
+     * current version is 2.5.0
+     */
+    function version() public override view returns (string memory){
+        return tokenVersion;
+    }
+
+    /**
+     * @dev Sets the values for `tokenName`, `tokenSymbol`, `tokenDecimals`,
+     * `tokenVersion` and `tokenOnchainID`
+     */
+
+    function setTokenInformation(string calldata _name, string calldata _symbol, uint8 _decimals, string calldata _version, address _onchainID) external override onlyOwner {
```

Comment:
I should be possible to set only one parameter,
setName, setSymbol, etc.

Reply:
Currently, we have a size problem because smart contracts are too big. It can be interesting once the size will be reduced